documents/coredocument.go

```diff
@@ -634,41 +634,41 @@ func (cd *CoreDocument) Timestamp() (time.Time, error) {
 }
 
 // AddAttribute adds a custom attribute to the model with the given value. If an attribute with the given name already exists, it's updated.
-func (cd *CoreDocument) AddAttribute(name string, attributeType attributeType, value string) (*CoreDocument, error) {
+func (cd *CoreDocument) AddAttribute(keyLabel string, attributeType attributeType, value string) (*CoreDocument, error) {
 	ncd, err := cd.PrepareNewVersion(nil, CollaboratorsAccess{})
 	if err != nil {
 		return nil, errors.NewTypedError(ErrCDAttribute, errors.New("failed to prepare new version: %v", err))
 	}
 	// TODO convert value from string to correct type
 	// For now its only string that is supported
-	nAttr, err := newAttribute(name, attributeType, value)
+	nAttr, err := newAttribute(keyLabel, attributeType, value)
```

Comment:
Currently, the `value` of the attribute is always of type `string`.

What would happen if I pass as attributeType of bytes. Where does the transformation happen?

Reply:
I know `newAttribute` can handle all types depending on the passed typ of value.

But currently the variable `value` would be always of type string but the passed attributeType my differ.